opentelemetry-sdk/src/resource/env.rs

```diff
@@ -104,7 +104,7 @@ mod tests {
             [
                 (
                     "OTEL_RESOURCE_ATTRIBUTES",
-                    Some("key=value, k = v , a= x, a=z"),
+                    Some("key=value, k = v , a= x, a=z,equal=value=,empty=,some=other=val"),
```

Comment:
Replace this with an example using properties. The primary reason for allowing the additional = is in support of this feature. 

https://www.w3.org/TR/baggage/#property